src/pages/DeviceSettings.tsx

```diff
@@ -0,0 +1,368 @@
+import { Router } from "@mui/icons-material";
+import { Box, Breadcrumbs, FormControl, Typography, MenuItem, Select, SelectChangeEvent } from "@mui/material";
+import { Link, useOutletContext, useParams } from "react-router-dom";
+import { useContext, useEffect, useState } from "react";
+import AddTextShow from "../components/shared/AddTextInput";
+import type { Device, } from "waziup";
+import { DevicesContext } from "../context/devices.context";
+import { devEUIGenerateFc, lineClamp, toStringHelper } from "../utils";
+import RowContainerBetween from "../components/shared/RowContainerBetween";
+import { DEFAULT_COLORS } from "../constants";
+// import { Android12Switch } from "../components/shared/Switch";
+export interface HTMLSelectProps extends React.SelectHTMLAttributes<HTMLSelectElement> {
+    handleChange: (event: SelectChangeEvent<string>) => void,
+    title: string,
+    conditions: string[] | number[],
+    value: string
+    isDisabled?: boolean
+    matches?: boolean
+    my?:number,
+    widthPassed?: string
+}
+export const SelectElement = ({ handleChange, title, conditions, isDisabled,my, widthPassed, name, value }: HTMLSelectProps) => (
+    <Box minWidth={120} width={widthPassed ? widthPassed : '100%'} my={my?my:.5}>
+        <Typography fontSize={12} fontWeight={'300'} color={'#292F3F'}>{title}</Typography>
+        <FormControl variant="standard" disabled={isDisabled} fullWidth>
+            <Select
+                inputProps={{
+                    name: name,
+                    id: 'uncontrolled-native',
+                }}
+                sx={{ fontWeight: 'bold' }}
+                value={value}
+                onChange={handleChange}
+            >
+                <MenuItem defaultChecked disabled value={''}>Select</MenuItem>
+                {
+                    conditions.map((condition, index) => (
+                        <MenuItem key={index} value={condition}>{condition}</MenuItem>
+                    ))
+                }
+            </Select>
+        </FormControl>
+    </Box>
+);
+import BoxDownload from '../assets/box_download.svg';
+import PrimaryButton from "../components/shared/PrimaryButton";
+import { DropDownCreateDeviceTab1 } from "../components/ui/CreateDeviceTab1";
+
+export default function DeviceSettings() {
+    function handleClick(event: React.MouseEvent<Element, MouseEvent>) {
+        event.preventDefault();
+        console.info('You clicked a breadcrumb.');
+    }
+    const { codecsList,devices,getDevicesFc,showDialog } = useContext(DevicesContext);
```

Comment:
I think that in this file, only the current device should be retrieved and stored.